Marlin/src/lcd/extensible_ui/ui_api.cpp

```diff
@@ -414,7 +421,11 @@ namespace UI {
   }
 
   void printFile(const char *filename) {
-    IFSD(card.openAndPrintFile(filename), NOOP);
+    #if ENABLED(SDSUPPORT)
+      card.openAndPrintFile(filename);
+    #else
+      NOOP;
+    #endif
```

Comment:
I bet this would work: `IFSD(card.openAndPrintFile(filename), 0);` because this is legal C/C++:
```cpp
  0;
```

Reply:
I thought I had tried that, and got a different compilation error or the method was optimised out or something I can not remember now, but I could have just done it wrong, I will give it a try now.